SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKCloudDeploymentConfiguration.cpp

```diff
@@ -1016,26 +1019,23 @@ FText SSpatialGDKCloudDeploymentConfiguration::GetSpatialOSRuntimeVersionToUseTe
 	return FText::FromString(RuntimeVersionString);
 }
 
-FReply SSpatialGDKCloudDeploymentConfiguration::OnGenerateConfigFromCurrentMap()
-{
-	UWorld* EditorWorld = GEditor->GetEditorWorldContext().World();
-	check(EditorWorld != nullptr);
-
-	const FString LaunchConfig = FPaths::Combine(FPaths::ConvertRelativePathToFull(FPaths::ProjectIntermediateDir()), FString::Printf(TEXT("Improbable/%s_CloudLaunchConfig.json"), *EditorWorld->GetMapName()));
 
-	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
-	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
-
-	FSpatialLaunchConfigDescription LaunchConfiguration = SpatialGDKEditorSettings->LaunchConfigDesc;
-	FWorkerTypeLaunchSection& ServerWorkerConfig = LaunchConfiguration.ServerWorkerConfig;
-
-	FillWorkerConfigurationFromCurrentMap(ServerWorkerConfig, LaunchConfiguration.World.Dimensions);
-
-	GenerateLaunchConfig(LaunchConfig, &LaunchConfiguration, ServerWorkerConfig);
+bool SSpatialGDKCloudDeploymentConfiguration::IsUsingAutoGenerateConfig() const
+{
+	const USpatialGDKEditorSettings* SpatialGKDSettings = GetDefault<USpatialGDKEditorSettings>();
+	return !SpatialGKDSettings->IsAutoGenerateConfigEnabled();
```

Comment:
Why is it checking that IsAutoGenerateConfigEnabled is false ?

Reply:
The naming of the function was wrong, it is used to determine whether the "Open Launch Configuration editor" button and the "Launch Config File" picker should be enabled or not.